static/app/views/explore/hooks/useExploreTimeseries.tsx

```diff
@@ -20,16 +21,94 @@ import {useSortedTimeSeries} from 'sentry/views/insights/common/queries/useSorte
 interface UseExploreTimeseriesOptions {
   enabled: boolean;
   query: string;
+  queryExtras?: {
+    fidelity?: 'low' | 'auto';
+  };
 }
 
 interface UseExploreTimeseriesResults {
   canUsePreviousResults: boolean;
   timeseriesResult: ReturnType<typeof useSortedTimeSeries>;
 }
 
-export function useExploreTimeseries({
+const LOW_FIDELITY_QUERY_EXTRAS = {
+  fidelity: 'low',
+} as const;
+
+const HIGH_FIDELITY_QUERY_EXTRAS = {
+  fidelity: 'auto',
+} as const;
+
+/**
+ * This hook is used to fetch timeseries data from the EAP dataset.
+ * It will trigger two queries, one that should resolve data quickly and
+ * one that will resolve more data but takes longer to execute.
+ *
+ * The hook will bias towards the high fidelity results if they are available.
+ * isLoading will be true if the high fidelity results are not available.
+ */
+export const useExploreTimeseries = ({
+  query,
+  enabled,
+}: {
+  enabled: boolean;
+  query: string;
+}) => {
+  const organization = useOrganization();
+  const canUseProgressiveLoading = organization.features.includes(
+    'visibility-explore-progressive-loading'
+  );
+
+  const {timeseriesResult, canUsePreviousResults} = useExploreTimeseriesImpl({
+    query,
+    enabled: enabled && !canUseProgressiveLoading,
+  });
+
+  // Start two queries with different fidelities, we will bias towards the high
+  // fidelity results if they are available
+  const {
+    timeseriesResult: lowFidelityTimeseriesResult,
+    canUsePreviousResults: canUsePreviousLowFidelityResults,
+  } = useExploreTimeseriesImpl({
+    query,
+    enabled: enabled && canUseProgressiveLoading,
+    queryExtras: LOW_FIDELITY_QUERY_EXTRAS,
+  });
+  const {
+    timeseriesResult: highFidelityTimeseriesResult,
+    canUsePreviousResults: canUsePreviousHighFidelityResults,
+  } = useExploreTimeseriesImpl({
+    query,
+    enabled: enabled && canUseProgressiveLoading,
```

Comment:
This will fire in parallel with the low fidelity request. Should there be some way to control if they're done in parallel or sequentially? Perhaps a prop like `queryMode: 'serial' | 'parallel'`.

And the condition can be

```
enabled && canUseProgressiveLoading && (queryMode === 'parallel' || lowFidelityTimeseriesResult.isLoading)
```

Alternatively, we can only permit serial loading to start.